keps/sig-windows/20190103-windows-node-support.md

```diff
@@ -130,16 +138,16 @@ As of 29-11-2018 much of the work for enabling Windows nodes has already been co
+**User experience**: Users today will need to use some combination of taints and node selectors in order to keep Linux and Windows workloads separated. In the best case this imposes a burden only on Windows users, but this is still less than ideal. The recommended approach is outlined below, with one of its main goals being that we should not break compatibility for existing Linux workloads
 
 #### Ensuring OS-specific workloads land on appropriate container host
-As you can see below, we plan to document how Windows containers can be scheduled on the appropriate host using Taints and Tolerations. All nodes today have the following default labels
+As you can see below, we plan to document how Windows containers can be scheduled on the appropriate host using Taints and Tolerations. All nodes today have the following default labels (These labels will be graduating to stable soon)
 - beta.kubernetes.io/os = [windows|linux]
 - beta.kubernetes.io/arch = [amd64|arm64|...]
 
 If a deployment does not specify a nodeSelector like `"beta.kubernetes.io/os": windows`, it is possible the Pods can be scheduled on any host, Windows or Linux. This can be problematic since a Windows container can only run on Windows and a Linux container can only run on Linux. The best practice we will recommend is to use a nodeSelector. 
 
-However, we understand that in certain cases customers have a pre-existing large number of deployments for Linux containers. Since they will not want to change all deployments to add nodeSelectors, the alternative is to use Taints. Because the kubelet can set Taints during registration, it could easily be modified to automatically add a taint when running on Windows only (`“--register-with-taints=’os=Win1809:NoSchedule’” `). By adding a taint to all Windows nodes, nothing will be scheduled on them (that includes existing Linux Pods). In order for a Windows Pod to be scheduled on a Windows node, it would need both the nodeSelector to choose Windows, and a toleration.
+However, we understand that in many cases customers have a pre-existing large number of deployments for Linux containers, as well as an ecosystem of off-the-shelf configurations, such as community Helm charts, and programmatic pod generation cases, such as with Operators. Customers will be hesitant to make the configuration change to add nodeSelectors. Our proposal as an alternative is to use Taints. Because the kubelet can set Taints during registration, it could easily be modified to automatically add a taint when running on Windows only (`“--register-with-taints=’os=Win1809:NoSchedule’” `). By adding a taint to all Windows nodes, nothing will be scheduled on them (that includes existing Linux Pods). In order for a Windows Pod to be scheduled on a Windows node, it would need both the nodeSelector to choose Windows, and a toleration.
```

Comment:
1809 is the same as Windows Server 2019. i used the 1809 designation since in the future we will support more releases like 1903, etc
Compatibility across different versions of container/OS is not our problem to solve. that comes directly from msft. i will mention this

Reply:
Does that mean windows pods have to target the exact version (1809, 1903, etc), instead of just windows server 2019?

I know it's beyond sig-windows' responsibility to guarantee compatibility, but it'd be nice to have some clarification (from Microsoft) and document this clearly.

Reply:
@yujuhong  yes that is correct. Windows PODs have to target a container host with the same exact version. Taints and tolerations have to be OS specific. In our case, and we documented that in the PR https://github.com/kubernetes/enhancements/pull/743, you need to create an OS specific taint.  For v1.14 only 1809 (windows server 2019) is supported, both as a container host and as the OS for a container.